import "goby/acomms/protobuf/driver_base.proto"; 
import "goby/acomms/protobuf/modem_message.proto";

import "dccl/option_extensions.proto";

package goby.acomms.store_server.protobuf;

message Config
{
    optional double query_interval_seconds = 1 [default = 1];
    optional int32 max_frame_size = 2 [default = 65536];
    optional double reset_interval_seconds = 3 [default = 120];
    repeated int32 rate_to_bytes = 4;
    repeated int32 rate_to_frames = 5;
}

extend goby.acomms.protobuf.DriverConfig
{
    optional Config config = 1321;
}

message Transmission
{
    enum TransmissionType
    {
        BASE_TYPE = 0;
        STORE_SERVER_DRIVER_POLL = 1;
    }
    optional TransmissionType type = 1321 [default = BASE_TYPE];
    
    // used in STORE_SERVER_DRIVER_POLL message
    optional int32 poll_src = 1 [(dccl.field).omit = true];
    optional int32 poll_dest = 2 [(dccl.field).omit = true];
}

extend goby.acomms.protobuf.ModemTransmission
{
    optional Transmission transmission = 1321;
}